### PR TITLE
fix: add ci for test, lint, format and typecheck

### PR DESCRIPTION
**Pull request by lstocchi**

### CODE REVIEW

**Comment by slemeur** [comment]
🚀

**Comment by feloy** [comment]
It seems you will have to run this command locally, to get some files correctly formatted (beware, files will be modified in place):

```
./node_modules/.bin/prettier -w --check "**/src/**/*.ts"
```

File: .github/workflows/pr-check.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+jobs:
+  lint-format-unit:
+    name: linter, formatters and unit tests / ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 40
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-2022, ubuntu-22.04]
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:30]: line 42 we see mac/Linux but here there is only Ubuntu, no macOS so I don't know if you omitted macOS on purpose or not

File: .github/workflows/pr-check.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        run: yarn typecheck
+
+      # Check we don't have changes in git
+      - name: Check no changes in git
+        if: ${{ matrix.os=='ubuntu-22.04'}}
+        run: |
+          if ! git diff --exit-code; then
+            echo "Found changes in git"
+            exit 1
+          fi
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:87]: looks like the file is not formatted (EOF missing)

File: package.json [hunk 3]
```diff
@@ -16,9 +16,16 @@
     "format:fix": "prettier --write \"**/src/**/*.ts\"",
     "lint:check": "eslint . --ext js,ts,tsx",
     "lint:fix": "eslint . --fix --ext js,ts,tsx",
-    "test:backend": "vitest run -r packages/backend --passWithNoTests --coverage"
+    "test:backend": "vitest run -r packages/backend --passWithNoTests --coverage",
+    "test:shared": "vitest run -r packages/shared --passWithNoTests --coverage",
```
Comment (benoitf) [inline package.json:20]: it is missing frontend tests, correct ?

**Review by benoitf — APPROVED** [review]

File: .github/workflows/pr-check.yaml (hunk 1 above)
Comment (lstocchi) [reply .github/workflows/pr-check.yaml:30]: i copied the portion i needed from the podman-desktop repo. I don't remember if there was some reasoning behind it. But i guess i can add the mac machine as well for what we do atm

Comment (lstocchi) [reply .github/workflows/pr-check.yaml]: Done 👍

File: packages/frontend/package.json [hunk 4]
```diff
@@ -29,10 +30,11 @@
     "@testing-library/user-event": "^14.5.1",
     "@tsconfig/svelte": "^5.0.2",
     "@typescript-eslint/eslint-plugin": "6.15.0",
+    "jsdom": "^23.2.0",
     "svelte": "4.2.8",
     "svelte-preprocess": "^5.1.3",
     "tailwindcss": "^3.4.0",
     "postcss": "^8.4.33",
-    "postcss-load-config": "^5.0.2"
+    "postcss-load-config": "^5.0.2"
```
Comment (benoitf) [inline packages/frontend/package.json:38]: it looks like the file is not formatted/sorted (usually it is ordered in alphabetical order) and there is an extra space here

Comment (lstocchi) [reply packages/frontend/package.json:38]: how do you sort it automatically? Is there any npm package? 🤔

Comment (feloy) [reply packages/frontend/package.json:38]: I think it is done when you run `yarn add <package>`

Comment (lstocchi) [reply packages/frontend/package.json:38]: I was hoping we could add some script so that if someone copy/paste an entry it is sorted anyway

Comment (benoitf) [reply packages/frontend/package.json]: `npx sort-package-json`

Comment (benoitf) [reply packages/frontend/package.json]: (anyway it's not a requested change)

Comment (lstocchi) [reply packages/frontend/package.json]: cool, i'm going to add it in a follow-up PR so i don't block @feloy 
for the moment i fixed it manually

**Pull request merged**